dd-trace/src/telemetry.rs

```diff
@@ -165,26 +173,61 @@ fn add_log_error_inner<I: Into<String>>(
     handle.add_error_log(message.into(), stack).ok();
 }
 
+pub fn notify_configuration_update(config_provider: &dyn ConfigurationProvider) {
+    notify_configuration_update_inner(config_provider, &TELEMETRY);
+}
+
+fn notify_configuration_update_inner(
+    config_provider: &dyn ConfigurationProvider,
+    telemetry_cell: &OnceLock<Arc<Mutex<Telemetry>>>,
+) {
+    let Some(telemetry) = telemetry_cell.get() else {
+        return;
+    };
+    let Ok(mut telemetry) = telemetry.lock() else {
+        return;
+    };
+    if !telemetry.enabled {
+        return;
+    }
+    let Some(handle) = telemetry.handle.as_mut() else {
+        return;
+    };
```

Comment:
Nit: you can replace this with `with_telemetry_handle` now that the other PR is merged.